TelCo.ColorCoder/pairnuberfromcolorpair.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Diagnostics;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+   class PairNumberEncoder
+   {
+       // Find the major color in the array and get the index
+    public static int CalculateMajorColorIndex(ColorPair pair)
+    {
```

Comment:
When modularizing, be sensitive to duplication. This function is very similar to CalculateMinorColorIndex... Think of a way to make it common.